### DISCUSSION TITLE
[✨] Receive `requestEvent` in `server$` differently

### DISCUSSION THREAD

**Discussion by niconiahi** [QwikDev/qwik-evolution#140] [Proposals For Qwik]
### Is your feature request related to a problem?

today, to access the `requestEvent` inside a clojure of a `server$` primitive, you have to use the _this_ keyword. I personally don't like using _this_ like that, feels so imperative. Where is it coming from? feels extra magical. I would prefer seeing it coming a parameter as all the other primitives do. That way _does_ feel really natural, because of the reactive (and functional IMO) nature of Qwik. The _this_ really feels like classes. So because of that I've thought that maybe it would be interesting to change how `requestEvent` is "handed off" from Qwik Router to `server$`

### Describe the solution you'd like

expectation:
```tsx
import { component$ } from '@builder.io/qwik';
import { server$, routeLoader$ } from '@builder.io/qwik/qwik-city';
 
const encryptInServer = server$(({ env }: RequestEvent) => (message: string, luckyNumber: number) => {
  const secretKey = env.get('SECRET_KEY');
  const encryptedMessage = encrypt(message, secretKey);
  console.log('the lucky number is =>', luckyNumber)
  return encryptedMessage;
});
 
export default component$(() => {
  return (
    <>
      <button
        onClick$={() => {
          const encrypted = await encryptInServer('Hello world', 22);
          console.log(encrypted);
        }}
      >
        Click me
      </button>
    </>
  );
});
```

as you control `server$`, I think it should be possible to take the parameters' values when executing `encryptInServer` (in the `onClick$`) and provide them in the `server$` as arguments but in a different way than today

<img width="1002" alt="Screenshot 2023-07-31 at 21 42 43" src="https://github.com/QwikDev/qwik-evolution/assets/8381041/ed9f4733-3698-476c-8b8d-858f7c84f4bd">

### Describe alternatives you've considered

I didn't consider any other alternative, to be honest

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
